snet_cli/commands.py

```diff
@@ -813,6 +812,102 @@ def _getserviceregistrationbyname(self, service_json=None):
             self._printerr("\nCall getServiceRegistrationByName() error!\nHINT: Check your identity and session.\n")
             self._error(e)
 
+    # Check if service json is valid
+    def _service_json_parser(self, service_json_path):
+        is_ok = True
+        service_json = None
+        try:
+            with open(service_json_path) as f:
+                service_json = json.load(f)
+        except Exception as e:
+            self._printerr("Failed to load {}".format(service_json_path))
+            self._error(e)
+
+        # Get list of valid service_spec/ files
+        valid_import_paths = set()
+        entry_path = None
+        if "service_spec" in service_json and service_json["service_spec"]:
+            spec_path = Path(service_json["service_spec"])
+            if not os.path.isabs(spec_path):
+                entry_path = Path.cwd().joinpath(spec_path).resolve()
+            else:
+                entry_path = spec_path.resolve()
+            if not os.path.isdir(entry_path):
+                self._printerr("Service spec path must resolve to a valid directory: {}".format(spec_path))
+                is_ok = False
+            import_paths = walk_imports(entry_path)
+
+            if import_paths:
+                codegen_tmp_dir = Path("~").expanduser().joinpath(".snet").joinpath("py-codegen").joinpath(
```

Comment:
BTW it seems this code duplicates https://github.com/singnet/snet-cli/blob/fdbaea78e566c6d470996a3243fe288f9532a94d/snet_cli/commands.py#L518

Reply:
I'll create a function to use this `protoc` part of code.